llvm/lib/Target/AMDGPU/AMDGPUSwLowerLDS.cpp

```diff
@@ -1218,6 +1218,8 @@ bool AMDGPUSwLowerLDS::run() {
     for (auto &K : FuncLDSAccessInfo.NonKernelToLDSAccessMap) {
       Function *Func = K.first;
       DenseSet<GlobalVariable *> &LDSGlobals = K.second;
+      if (Func->isDeclaration() || LDSGlobals.empty())
```

Comment:
How would a declaration end up in this set to begin with? Does this ever fire?

Reply:
Fixed this in latest commit. Skipped adding function to map if its a declaration.